### PR TITLE
python311Packages.scs: fix build; 3.2.3 -> 3.2.4

### PR DESCRIPTION
**Pull request by ghost**
numpy change 1.25.1 -> 1.26.1  237a5900169af2ce9b56ac1e2e77575fcaf9310c  moved away from distutils and the scs build was unable to locate the blas and lapack libs and failed.

this change fixes the build by adding env vars so the build can locate the blas lapack libs for linux and the Accelerate lib for darwin. scs is then updated to 3.2.4. incorporates upstream PR https://github.com/bodono/scs-python/pull/78 to use pkg-config and add myself as maintainer so i can remove the patches on the next version update.

Note that the linear algebra library for darwin is Accelerate as that is prioritized over blas / lapack by the build.

https://hydra.nixos.org/build/241138404
ZHF: #265948

bambi failures due to numpy < 1.26 constraints in pytensor

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>28 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.qiskit</li>
    <li>python310Packages.qiskit-aer</li>
    <li>python310Packages.qiskit-aer.dist</li>
    <li>python310Packages.qiskit-finance</li>
    <li>python310Packages.qiskit-finance.dist</li>
    <li>python310Packages.qiskit-ibmq-provider</li>
    <li>python310Packages.qiskit-ibmq-provider.dist</li>
    <li>python310Packages.qiskit-ignis</li>
    <li>python310Packages.qiskit-ignis.dist</li>
    <li>python310Packages.qiskit-nature</li>
    <li>python310Packages.qiskit-nature.dist</li>
    <li>python310Packages.qiskit-optimization</li>
    <li>python310Packages.qiskit-optimization.dist</li>
    <li>python310Packages.qiskit.dist</li>
    <li>python311Packages.qiskit</li>
    <li>python311Packages.qiskit-aer</li>
    <li>python311Packages.qiskit-aer.dist</li>
    <li>python311Packages.qiskit-finance</li>
    <li>python311Packages.qiskit-finance.dist</li>
    <li>python311Packages.qiskit-ibmq-provider</li>
    <li>python311Packages.qiskit-ibmq-provider.dist</li>
    <li>python311Packages.qiskit-ignis</li>
    <li>python311Packages.qiskit-ignis.dist</li>
    <li>python311Packages.qiskit-nature</li>
    <li>python311Packages.qiskit-nature.dist</li>
    <li>python311Packages.qiskit-optimization</li>
    <li>python311Packages.qiskit-optimization.dist</li>
    <li>python311Packages.qiskit.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python310Packages.bambi</li>
    <li>python310Packages.bambi.dist</li>
    <li>python311Packages.bambi</li>
    <li>python311Packages.bambi.dist</li>
  </ul>
</details>
<details>
  <summary>24 packages built:</summary>
  <ul>
    <li>python310Packages.blackjax</li>
    <li>python310Packages.blackjax.dist</li>
    <li>python310Packages.cvxpy</li>
    <li>python310Packages.cvxpy.dist</li>
    <li>python310Packages.jaxopt</li>
    <li>python310Packages.jaxopt.dist</li>
    <li>python310Packages.qpsolvers</li>
    <li>python310Packages.qpsolvers.dist</li>
    <li>python310Packages.qutip</li>
    <li>python310Packages.qutip.dist</li>
    <li>python310Packages.scs</li>
    <li>python310Packages.scs.dist</li>
    <li>python311Packages.blackjax</li>
    <li>python311Packages.blackjax.dist</li>
    <li>python311Packages.cvxpy</li>
    <li>python311Packages.cvxpy.dist</li>
    <li>python311Packages.jaxopt</li>
    <li>python311Packages.jaxopt.dist</li>
    <li>python311Packages.qpsolvers</li>
    <li>python311Packages.qpsolvers.dist</li>
    <li>python311Packages.qutip</li>
    <li>python311Packages.qutip.dist</li>
    <li>python311Packages.scs</li>
    <li>python311Packages.scs.dist</li>
  </ul>
</details>

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by a-n-n-a-l-e-e** [comment]
looking at pkgs/development/python-modules/numpy/default.nix it seems that site.cfg with the lapack and blas is written but not installed by the build. perhaps this is a numpy build issue.

**Comment by a-n-n-a-l-e-e** [comment]
darwin [failing](https://logs.ofborg.org/?key=nixos/nixpkgs.268178&attempt_id=3a899557-31f0-4eb4-b95f-47e6410d98a4)
```
Blas info:
{'extra_compile_args': ['-msse3', '-I/System/Library/Frameworks/vecLib.framework/Headers'], 'extra_link_args': ['-Wl,-framework', '-Wl,Accelerate'], 'define_macros': [('NO_ATLAS_INFO', 3), ('HAVE_CBLAS', None), ('ACCELERATE_NEW_LAPACK', None)]}
Lapack info:
{'extra_compile_args': ['-msse3', '-I/System/Library/Frameworks/vecLib.framework/Headers'], 'extra_link_args': ['-Wl,-framework', '-Wl,Accelerate'], 'define_macros': [('NO_ATLAS_INFO', 3), ('HAVE_CBLAS', None), ('ACCELERATE_NEW_LAPACK', None)]}
...
ld: framework not found Accelerate
clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
```

hydra logs: https://hydra.nixos.org/build/241080821 last success before numpy change  [log](https://hydra.nixos.org/log/ycq6aprgy936ic6sv36grrbj2bxvag4c-python3.11-scs-3.2.3.drv)

is not trying to link with `Accelerate`. 
```
Blas info:
{'libraries': ['lapack', 'lapacke', 'blas', 'cblas', 'lapack', 'lapacke', 'blas', 'cblas'], 'library_dirs': ['/nix/store/ax4g6bx0c9q6qss26rp0hxmilzpz1iix-lapack-3/lib', '/nix/store/ssyzhlg1hy50rkzcl7b7ii375r2a3bw1-blas-3/lib'], 'language': 'c', 'define_macros': [('HAVE_CBLAS', None)], 'runtime_library_dirs': ['/nix/store/ax4g6bx0c9q6qss26rp0hxmilzpz1iix-lapack-3/lib', '/nix/store/ssyzhlg1hy50rkzcl7b7ii375r2a3bw1-blas-3/lib']}
Lapack info:
{'libraries': ['lapack', 'lapacke', 'blas', 'cblas', 'lapack', 'lapacke', 'blas', 'cblas'], 'library_dirs': ['/nix/store/ax4g6bx0c9q6qss26rp0hxmilzpz1iix-lapack-3/lib', '/nix/store/ssyzhlg1hy50rkzcl7b7ii375r2a3bw1-blas-3/lib'], 'language': 'c', 'define_macros': [('HAVE_CBLAS', None)], 'runtime_library_dirs': ['/nix/store/ax4g6bx0c9q6qss26rp0hxmilzpz1iix-lapack-3/lib', '/nix/store/ssyzhlg1hy50rkzcl7b7ii375r2a3bw1-blas-3/lib']}
```

**Comment by a-n-n-a-l-e-e** [comment]
adding Accelerate to buildInputs seemed to work -- but just realized that a new scs release is out that uses meson so trying to get that to work in #268228

**Comment by mweinelt** [comment]
Annoying that they don't use pkg-config for that.

**Comment by a-n-n-a-l-e-e** [comment]
cc @renesat  given past changes

**Comment by a-n-n-a-l-e-e** [comment]
> Annoying that they don't use pkg-config for that.

yeah -- i don't know meson well, but changing cc.find_library -> dependency and adding pkg-config works well on linux.

**Comment by mweinelt** [comment]
That might be a good thing to propose upstream, if you're interested.

**Comment by a-n-n-a-l-e-e** [comment]
> That might be a good thing to propose upstream, if you're interested.

proposed -- doesn't work with there CI for some reason. I was hoping dependency would fall back to find_library, but it doesn't seem to.

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -1225,6 +1225,10 @@
     githubId = 81317317;
     name = "Anna Aurora";
   };
+  a-n-n-a-l-e-e = {
+    github = "a-n-n-a-l-e-e";
+    githubId = 150648636;
+  };
```
Comment (mweinelt) [inline maintainers/maintainer-list.nix:1231]: As per the header in this file, the `name` field is required.

**Comment by mweinelt** [comment]
Awesome work!

**Pull request merged**